src/NuGet.Core/NuGet.Configuration/Settings/SettingFactory.cs

```diff
@@ -176,7 +176,7 @@ public int GetHashCode(SettingElement obj)
             if (xElement.Name.LocalName.Equals(ConfigurationConstants.PackageSourceMapping, StringComparison.OrdinalIgnoreCase) && duplicatedDescendants != null)
             {
                 var duplicatedKey = string.Join(", ", duplicatedDescendants.Select(d => d.Attributes["key"]));
-                var source = duplicatedDescendants.Select(d => d.Origin.ConfigFilePath).First();
+                var source = duplicatedDescendants.Select(d => d.Origin?.ConfigFilePath).Where(d => d is not null).First();
```

Comment:
Not a hot path so it doesn't really matter but this might as well just call `First()`

```suggestion
                var source = duplicatedDescendants.Select(d => d.Origin?.ConfigFilePath).First(d => d is not null);
```